### PR TITLE
Allow to disable background cache download for loading metadata files

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add a settings `filesystem_cache_enable_background_download_for_metadata_files` for filesystem cache to allow to disable background download of filesystem cache for metadata files. This feature is needed for private code feature, for public version it does not make much sense.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [x]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: src/Core/Settings.cpp
Comment (kssenii) [inline src/Core/Settings.cpp:1]: A test for this feature will be added in the Sync PR.

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit f482e43a75c10b504faf04525798b89d80c28918 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70806/f482e43a75c10b504faf04525798b89d80c28918/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/fce19cd318d1dc152fb5f91feed71cf2533607c5/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by jkartseva** [comment]
`02789_filesystem_cache_alignment` looks related: https://s3.amazonaws.com/clickhouse-test-reports/70806/c97c6250fcdcd6059753738bd12928a1e3fb2ac7/stateless_tests__asan__[1_2].html

**Comment by kssenii** [comment]
> 02789_filesystem_cache_alignment looks related

it got fixed after syncing changes from Sync PR, the pubilc version was lacking the fixed I did in private

**Comment by kssenii** [comment]
> Stateless tests

02962_system_sync_replica_lightweight_from_modifier - [flaky](https://play.clickhouse.com/play?user=play#CldJVEgKICAgICcwMjk2Ml9zeXN0ZW1fc3luY19yZXBsaWNhX2xpZ2h0d2VpZ2h0X2Zyb21fbW9kaWZpZXInIEFTIG5hbWVfc3Vic3RyLAogICAgOTAgQVMgaW50ZXJ2YWxfZGF5cywKICAgICgnU3RhdGVsZXNzIHRlc3RzIChhc2FuKScsICdTdGF0ZWxlc3MgdGVzdHMgKGFkZHJlc3MpJywgJ1N0YXRlbGVzcyB0ZXN0cyAoYWRkcmVzcywgYWN0aW9ucyknKSBBUyBiYWNrcG9ydF9hbmRfcmVsZWFzZV9zcGVjaWZpY19jaGVja3MKU0VMRUNUCiAgICB0ZXN0X25hbWUsCiAgICB0b1N0YXJ0T2ZEYXkoY2hlY2tfc3RhcnRfdGltZSkgQVMgZCwKICAgIGNvdW50KCksCiAgICBncm91cFVuaXFBcnJheShwdWxsX3JlcXVlc3RfbnVtYmVyKSBBUyBwcnMsCiAgICBhbnkocmVwb3J0X3VybCkKRlJPTSBjaGVja3MKV0hFUkUgKChub3coKSAtIHRvSW50ZXJ2YWxEYXkoaW50ZXJ2YWxfZGF5cykpIDw9IGNoZWNrX3N0YXJ0X3RpbWUpIEFORCAocHVsbF9yZXF1ZXN0X251bWJlciBOT1QgSU4gKAogICAgU0VMRUNUIHB1bGxfcmVxdWVzdF9udW1iZXIgQVMgcHJuCiAgICBGUk9NIGNoZWNrcwogICAgV0hFUkUgKHBybiAhPSAwKSBBTkQgKChub3coKSAtIHRvSW50ZXJ2YWxEYXkoaW50ZXJ2YWxfZGF5cykpIDw9IGNoZWNrX3N0YXJ0X3RpbWUpIEFORCAoY2hlY2tfbmFtZSBJTiAoYmFja3BvcnRfYW5kX3JlbGVhc2Vfc3BlY2lmaWNfY2hlY2tzKSkKKSkgQU5EIChwb3NpdGlvbih0ZXN0X25hbWUsIG5hbWVfc3Vic3RyKSA+IDApIEFORCAodGVzdF9zdGF0dXMgSU4gKCdGQUlMJywgJ0VSUk9SJywgJ0ZMQUtZJykpCkdST1VQIEJZIGQsIHRlc3RfbmFtZQpPUkRFUiBCWSBkIERFU0MK)
00653_verification_monotonic_data_load - [flaky](https://play.clickhouse.com/play?user=play#CldJVEgKICAgICcwMDY1M192ZXJpZmljYXRpb25fbW9ub3RvbmljX2RhdGFfbG9hZCcgQVMgbmFtZV9zdWJzdHIsCiAgICA5MCBBUyBpbnRlcnZhbF9kYXlzLAogICAgKCdTdGF0ZWxlc3MgdGVzdHMgKGFzYW4pJywgJ1N0YXRlbGVzcyB0ZXN0cyAoYWRkcmVzcyknLCAnU3RhdGVsZXNzIHRlc3RzIChhZGRyZXNzLCBhY3Rpb25zKScpIEFTIGJhY2twb3J0X2FuZF9yZWxlYXNlX3NwZWNpZmljX2NoZWNrcwpTRUxFQ1QKICAgIHRlc3RfbmFtZSwKICAgIHRvU3RhcnRPZkRheShjaGVja19zdGFydF90aW1lKSBBUyBkLAogICAgY291bnQoKSwKICAgIGdyb3VwVW5pcUFycmF5KHB1bGxfcmVxdWVzdF9udW1iZXIpIEFTIHBycywKICAgIGFueShyZXBvcnRfdXJsKQpGUk9NIGNoZWNrcwpXSEVSRSAoKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZSkgQU5EIChwdWxsX3JlcXVlc3RfbnVtYmVyIE5PVCBJTiAoCiAgICBTRUxFQ1QgcHVsbF9yZXF1ZXN0X251bWJlciBBUyBwcm4KICAgIEZST00gY2hlY2tzCiAgICBXSEVSRSAocHJuICE9IDApIEFORCAoKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZSkgQU5EIChjaGVja19uYW1lIElOIChiYWNrcG9ydF9hbmRfcmVsZWFzZV9zcGVjaWZpY19jaGVja3MpKQopKSBBTkQgKHBvc2l0aW9uKHRlc3RfbmFtZSwgbmFtZV9zdWJzdHIpID4gMCkgQU5EICh0ZXN0X3N0YXR1cyBJTiAoJ0ZBSUwnLCAnRVJST1InLCAnRkxBS1knKSkKR1JPVVAgQlkgZCwgdGVzdF9uYW1lCk9SREVSIEJZIGQgREVTQwo=)

> AST fuzzer (ubsan) 

https://github.com/ClickHouse/ClickHouse/issues/56640

**Review by jkartseva — APPROVED** [review]

**Comment by kssenii** [comment]
> Integration tests

`test_quorum_inserts/test.py::test_insert_quorum_with_keeper_loss_connection`

> CH Inc sync — Failed. Needs manual intervention.

I fixed it, check status not up-to-date

**Pull request merged**